### PR TITLE
Allure 2

### PR DESCRIPTION
**Pull request by andrcuns**
This is a complete rewrite of commons adapter for allure 2.0+ support.
It follows similar architecture of how java and js adaptors are implemented.

# Limitations
This is in no way, shape or form compatible with previous version of this adapter.
Currently nested steps are not supported, a bit more sophisticated store for steps would need to be implemented similar how it is done in java adapter.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,45 +1,47 @@
...
+## Configuration
 
-You can specify the directory where the Allure test results will appear. By default it would be 'gen/allure-results' within
-your current directory.
+Following configuration options are supported:
 
 ```ruby
-    AllureRubyAdaptorApi.configure do |c|
+    Allure.configure do |c|
       c.output_dir = "/whatever/you/like"
```
Comment (baev) [inline README.md]: results_directory

File: README.md [hunk 2]
```diff
@@ -1,45 +1,47 @@
...
+      c.logging_level = Logger::INFO
+      # these are used for creating links to bugs or test cases where {} is replaced with keys of relevant items
+      c.tms_link_pattern = "http://www.jira.com/browse/{}"
```
Comment (baev) [inline README.md]: link_tms_pattern

File: README.md [hunk 3]
```diff
@@ -1,45 +1,47 @@
...
+      c.issue_link_pattern = "http://www.jira.com/browse/{}"
```
Comment (baev) [inline README.md]: link_issue_pattern

File: lib/allure-ruby-commons.rb [hunk 4]
```diff
@@ -0,0 +1,50 @@
+# rubocop:disable Naming/FileName
+# frozen_string_literal: true
+
+require "require_all"
+require "uuid"
+require_rel "allure_ruby_commons/**/*rb"
+
+# Namespace for classes that handle allure report generation and different framework adapters
+module Allure
```
Comment (baev) [inline lib/allure-ruby-commons.rb]: can we also add all non-deprecated methods from https://github.com/allure-framework/allure-java/blob/master/allure-java-commons/src/main/java/io/qameta/allure/Allure.java

File: lib/allure_ruby_commons/model/status.rb [hunk 5]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+module Allure
+  class Status
```
Comment (baev) [inline lib/allure_ruby_commons/model/status.rb]: we only have 4 statuses

https://github.com/allure-framework/allure-java/blob/master/allure-model/src/main/java/io/qameta/allure/model/Status.java

File: lib/allure-ruby-commons.rb (hunk 4 above)
Comment (andrcuns) [reply lib/allure-ruby-commons.rb]: Added all except `step`. Will add that in a separate PR as it requires adding support for nested steps to implement it properly.

**Comment by baev** [comment]
changes moved to separate repository https://github.com/allure-framework/allure-ruby

**Pull request closed without merging**